### PR TITLE
Voting material

### PR DESCRIPTION
**Pull request by JDawes-ScottLogic**
Initial work on updating the voting page to use angular material.

:warning:This breaks the existing voting/results pages:warning:

Added basic, multi, up/down poll types. Points is still in progress.
NamedVoting is implemented.
Expirydate is partially implemented, but needs formatting.
User choices are not implemented.
Results/manage link are not implemented (and may not be).
Errors are not currently handled (Toasts).

Tests have not been migrated yet, and old code still remains.

### CODE REVIEW

File: VotingApplication/VotingApplication.Web/Scripts/Directives/VoteChoice/VoteChoice.html [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+    <div ng-switch-when="Multi">
+        <multi-vote-choice choices="choices"></multi-vote-choice>
+    </div>
+
+    <div ng-switch-when="UpDown">
+        <up-down-vote-choice choices="choices"></up-down-vote-choice>
+    </div>
+
+    <div ng-switch-when="Points">POINTS</div>
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web/Scripts/Directives/VoteChoice/VoteChoice.html]: Is this a ToDo?

File: VotingApplication/VotingApplication.Web/Scripts/Modules/Poll/VoteOn-Poll.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+﻿(function () {
+    'use strict';
+
+    angular
+        .module('VoteOn-Poll', ['VoteOn-Common', 'VoteOn-Account', 'VoteOn-Vote'])
+            .config([
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web/Scripts/Modules/Poll/VoteOn-Poll.js]: Just a small formatting note; we typically indent config/controller/directive etc. the same amount as the module.
i.e. .module and .config have the same indentation.

File: VotingApplication/VotingApplication.Web/Scripts/Modules/VoteOn-Common.js [hunk 3]
```diff
@@ -2,7 +2,7 @@
     'use strict';
 
     angular
-        .module('VoteOn-Common', [])
+        .module('VoteOn-Common', ['ngMaterial', 'ngMessages'])
             .config(function ($mdThemingProvider) {
```
Comment (stevenhillcox) [inline VotingApplication/VotingApplication.Web/Scripts/Modules/VoteOn-Common.js]: Same formatting note as above

File: VotingApplication/VotingApplication.Web/Scripts/Directives/VoteChoice/VoteChoice.html (hunk 1 above)
Comment (JDawes-ScottLogic) [inline VotingApplication/VotingApplication.Web/Scripts/Directives/VoteChoice/VoteChoice.html]: Yup. Tried something with sliders, but the angular material sliders don't currently do what we want (partial disablement). As noted in the PR comment, Points are are still in progress, as I'm not sure what they're going to look like yet...

**Pull request merged**